### ISSUE TITLE
SpectatorViewPlugin, "method not implemented" with Unity 2018.2

### ISSUE THREAD

**Issue by cdsenselab** [Microsoft/MixedRealityToolkit-Unity#2919]
Hi!

I'm trying to get the SpectatorViewPlugin to work with Unity 2018.2, but I'm stuck with the following error:
```
error:14090086:lib(20):func(144):reason(134)
error:14090086:lib(20):func(144):reason(134)
'MarkerDetection.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\MarkerDetectionVS.Debug_x86.cd\System.Diagnostics.StackTrace.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Exception: The method or operation is not implemented.
Type: System.NotImplementedException
Module: WinRTBridge
InnerException: <No Data>
AdditionalInfo:MarshalFirstFieldIntoNativePtr type: UnityEngine.XR.WSA.WebCam.PhotoCaptureFrame
   at WinRTBridge.Marshalling.MarshalFirstFieldIntoNativePtr(Object instance)
 
(Filename:  Line: 0)


ArgumentNullException: Value cannot be null.
Parameter name: _unity_self
   at UnityEngineProxy.InternalCalls.PhotoCaptureFrame_CUSTOM_GetDataLength(Object self)
   at UnityEngine.XR.WSA.WebCam.PhotoCaptureFrame..ctor(IntPtr nativePtr)
   at UnityEngine.XR.WSA.WebCam.PhotoCapture.InvokeOnCapturedPhotoToMemoryDelegate(OnCapturedToMemoryCallback callback, Int64 hResult, IntPtr photoCaptureFramePtr)
   at UnityEngine.XR.WSA.WebCam.PhotoCapture.$Invoke10InvokeOnCapturedPhotoToMemoryDelegate(Int64 instance, Int64* args)
   at UnityEngine.Internal.$MethodUtility.InvokeMethod(Int64 instance, Int64* args, IntPtr method) 
(Filename: <Unknown> Line: 0)
```
Any ideas? 

I could build the SpectatorViewPlugin as described in https://github.com/Microsoft/MixedRealityToolkit/tree/master/SpectatorViewPlugin and I put the DLLs into the folder Assets/Plugins/WSA/x86/. The DLLs must be found, without them I got the expected DLL missig errors. I'm using Unity 2018.2.5f1 and also tried 2018.2.12f1, with the same error.

Best,
   Claus

**Comment by isurfraz** [comment]
Can someone give some official response on this? This is a great feature that our clients want but currently does not work with the latest official release of unity, starting to get very tired of having to navigate back here for every useful part of the toolkit only to find  that many users are experiencing issues with them

**Comment by Vernimator** [comment]
I too would like to implement this in my project but I cannot get the plugin to build. I get the following error.

1>------ Build started: Project: SpectatorViewPlugin, Configuration: Release Win32 ------
1>Object reference not set to an instance of an object.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

Here is a link to my post on the error.

https://github.com/Microsoft/MixedRealityToolkit/issues/182

**Comment by tvald** [comment]
Reported on the Unity WMR forum: https://forum.unity.com/threads/system-notimplementedexception-when-a-photocaptureframe-instantiated.576460/

**Comment by tvald** [comment]
Fixed in Unity 2018.3:
https://issuetracker.unity3d.com/issues/windowsmr-failure-to-take-photo-capture-in-hololens
https://forum.unity.com/threads/hololens-photo-capturing-failing.548845/#post-3826153

**Comment by timenick** [comment]
> I too would like to implement this in my project but I cannot get the plugin to build. I get the following error.
> 
> 1>------ Build started: Project: SpectatorViewPlugin, Configuration: Release Win32 ------
> 1>Object reference not set to an instance of an object.
> ========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
> 
> Here is a link to my post on the error.
> 
> [Microsoft/MixedRealityToolkit#182](https://github.com/Microsoft/MixedRealityToolkit/issues/182)

The following branch worked for me. Please have a try.

https://github.com/Microsoft/vcpkg/tree/dev/roschuma/4574

**Comment by ras0219-msft** [comment]
>https://github.com/Microsoft/vcpkg/tree/dev/roschuma/4574

This branch has been merged into vcpkg master, so please use that instead!

**Comment by keveleigh** [comment]
Closing, as fixes were reported above in both Unity 2018.3  and the vcpkg repo. Please reopen if not.
